### ISSUE TITLE
CPS e00900p + e00900s deviates from e00900 after advancing calculator

### ISSUE THREAD

**Issue by MaxGhenis** [open-source-economics/Tax-Calculator#2024]
`Records.cps_constructor` asserts that `e00900 = e00900p + e00900s` for all records. This is true in the raw CPS, but after advancing, it no longer holds. Deviations are relatively small (largest is $11) but this is enough to break `cps_constructor` if feeding back advanced data into the function (rationale in https://github.com/open-source-economics/Tax-Calculator/issues/2025). [Here's the notebook.](http://nbviewer.jupyter.org/github/MaxGhenis/taxcalc-notebooks/blob/master/exploratory/e00900p_e00900s_growth.ipynb)

**Comment by MaxGhenis** [comment]
I think this could be resolved by replacing https://github.com/open-source-economics/Tax-Calculator/blob/master/taxcalc/records.py#L341, which blows up `e00900`, with a line that sets `e00900` to `e00900p + e00900s` after those two are blown up. The same could be done for other variables with `p`/`s` components: `e00200` and `e02100`.

If this sounds right I can put a PR in.

**Comment by MaxGhenis** [comment]
Actually `e00900` has unique potential to differ because each component could be multiplied by different factors. This is probably occurring within records where `np.sign(e00900p) != np.sign(e00900s)`.

```
self.e00900[:] = np.where(self.e00900 >= 0,  # Same for p/s.
                          self.e00900 * ASCHCI,
                          self.e00900 * ASCHCL)
```

**Comment by martinholmer** [comment]
Thanks for the bug report. 

I’ll work on this Tuesday when I return to the office

**Comment by martinholmer** [comment]
A proposed fix for this bug is included in pull request #2027.
@MaxGhenis, is this the fix you had in mind?

**Comment by MaxGhenis** [comment]
Yes, that looks right. Thanks!

**Comment by martinholmer** [comment]
The bug reported in this issue has been fixed in pull request #2027.
